components/annotationList.js

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import Router from "next/router";
+import { render } from "react-dom";
+
+/**
+ * annotations(annArray) : function that returns 1 array of all the annotation tags
+ *
+ * @param {*} annArray : array of annotations (which each annotation contains an array of tags)
+ */
+function annotations(annArray) {
+  if (Array.isArray(annArray)) {
+    var returnArray = [];
+    return annArray.map(x => returnArray.concat(x.Tags)).flat();
+  } else {
+    console.log(annArray);
+  }
+}
+
+function mapAnnotationsHelper(uniqueArray) {
+  let rows = [];
+  for (var j = 0; j < uniqueArray.length; j++) {
+    rows.push(
+      <div>
+        <a href={"/search/" + uniqueArray[j]}>{uniqueArray[j]}</a>
```

Comment:
Here is a great example of using map if you want :)
`return uniqueArray.map (item =>  <div><a href={"/search/" + item }>{item }</a> </div>`
so this function is now only one line of code.